sdk/appplatform/mgmt/src/test/java/com/azure/resourcemanager/appplatform/SpringCloudTest.java

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.resourcemanager.appplatform;
+
+import com.azure.resourcemanager.appplatform.models.RuntimeVersion;
+import com.azure.resourcemanager.appplatform.models.SpringApp;
+import com.azure.resourcemanager.appplatform.models.SpringAppDeployment;
+import com.azure.resourcemanager.appplatform.models.SpringService;
+import com.azure.resourcemanager.resources.fluentcore.arm.Region;
+import com.azure.resourcemanager.resources.fluentcore.utils.SdkContext;
+import org.apache.commons.compress.utils.IOUtils;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.net.URLConnection;
+
+public class SpringCloudTest extends AppPlatformTest {
+
+    SpringCloudTest() {
+        super(RunCondition.LIVE_ONLY); // need storage data-plane and url check
+    }
+
+    @Test
+    public void canCRUDSpringAppWithDeployment() throws IOException {
+        String serviceName = generateRandomResourceName("springsvc", 15);
+        String appName = "gateway";
+        String deploymentName = generateRandomResourceName("deploy", 15);
+        String deploymentName1 = generateRandomResourceName("deploy", 15);
+
+        SpringService service = appPlatformManager.springServices().define(serviceName)
+            .withRegion(Region.US_EAST)
+            .withNewResourceGroup(rgName)
+            .withSku("B0")
+            .create();
+
+        Assertions.assertEquals("B0", service.sku().name());
+
+        service.update()
+            .withSku("S0")
+            .apply();
+
+        Assertions.assertEquals("S0", service.sku().name());
+
+        service.update()
+            .withGitUri("https://github.com/Azure-Samples/piggymetrics-config")
+            .apply();
+        Assertions.assertEquals("https://github.com/Azure-Samples/piggymetrics-config", service.serverProperties().configServer().gitProperty().uri());
+
+        File jarFile = new File("gateway.jar");
+        if (!jarFile.exists()) {
+            URLConnection connection = new URL("https://github.com/weidongxu-microsoft/azure-sdk-for-java-management-tests/raw/master/spring-cloud/gateway.jar").openConnection();
+            connection.connect();
+            IOUtils.copy(connection.getInputStream(), new FileOutputStream(jarFile));
```

Comment:
I think you need to close `connection.getInputStream()`

Maybe close `FileOutputStream` as well.